src/PasswordHasher.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+interface PasswordHasher {
+
+    public function hashAndLoginPassword($password, $row);
+
+    public function hashAndRegisterPassword($password);
```

Comment:
Funkcijų pavadinimai šiame interface drįsčiau sakyt net klaidinantys. Ar ši tavo funkcija užhashuoja ir paskui kažkur užregistruoja slaptažodį? Nes pagal pavadinimą tikrai tikėčiausi to. Siūlau vieną metodą turėti hashPassword, o kitą - passwordsEqual kuris kaip argumentus priimtų password, hash, salt, hashTimes, o grąžintų true/false. Galbūt reiktų pakeisti vartotojo prijungimo logiką, bet bent jau neklaidintum kolegų programuotojų